include/sys/util_macro.h

```diff
@@ -41,12 +43,12 @@ extern "C" {
  * @brief Unsigned integer with bit position @p n set (signed in
  * assembly language).
  */
-#define BIT(n)  (1UL << (n))
+#define BIT(n)  ((uint32_t)1 << (unsigned int)(n))
```

Comment:
This changes `warning: left shift count is negative` to `warning: left shift count >= width of type`
How does `(unsigned int)(n)` improve our code overall?

Reply:
This part is indeed my nightmare as this file is used by assembler, dtc and compiler.
I've now tried a different solution to keep BIT macro unchanged.

Reply:
Hmm. Are we sure that we don't use the result of BIT() macro somewhere in ASM code?

Reply:
You are looking at an obsolete version.